### ISSUE TITLE
Finalization phase of the install fails due to missing /run/lock

### ISSUE THREAD

**Issue by msdobrescu** [mocaccinoOS/os-commons#23]
At the end of the system install performed by Calamares, the following packages are removed:

https://github.com/msdobrescu/os-commons/blob/41301bfbfd49ee21dd00f235882eb3c854be31b4/packages/mocaccino/mocaccino-calamares/mocaccino/main.py#L17

The operation fails due to missing /run/lock on the target system.

Adding that right before the removal would succeed:

```
...
# create /run/lock
libcalamares.utils.target_env_call(["mkdir", "-p", "/run/lock"])

if len(luet_packages2remove) > 0:
        args = ["luet", "uninstall", "-y"]
        # args = args + luet_packages2remove
        # libcalamares.utils.target_env_call(args)
        # Temporary trying to remove every package singolary
        for pkg in luet_packages2remove:
            libcalamares.utils.target_env_call(args + [pkg])

...
```

On the target system, /var/lock already exists and points to /run/lock.

**Comment by msdobrescu** [comment]
[calamares-FAILS.log](https://github.com/mocaccinoOS/os-commons/files/7190328/calamares-FAILS.log)
[calamares-OK.log](https://github.com/mocaccinoOS/os-commons/files/7190329/calamares-OK.log)

**Comment by geaaru** [comment]
Thanks for the report.
It seems related to this: https://forums.gentoo.org/viewtopic-t-1089460-start-0.html

@joostruis I think that you can to add the package `'sys-apps/gentoo-systemd-integration` or create the needed file to set under the directory `/usr/lib/tmpfiles.d/`

**Comment by joostruis** [comment]
Just FYI we have sys-apps/gentoo-systemd-integration installed.

**Comment by msdobrescu** [comment]
> Thanks for the report.
> It seems related to this: https://forums.gentoo.org/viewtopic-t-1089460-start-0.html
> 
> @joostruis I think that you can to add the package `'sys-apps/gentoo-systemd-integration` or create the needed file to set under the directory `/usr/lib/tmpfiles.d/`

```
The package and the config file to create the filesystem exists as part of the live system in `/usr/lib/tmpfiles.d/gentoo-run.conf`:

# New OpenRC versions replace /var/lock with a symlink to /run/lock.
# We just create that directory for now, just in case.
# Of course, it's not guaranteed that'll satisfy all relevant packages.

d /run/lock 0775 root uucp -
```

Testing shows that the above file is deployed too, so, presumably, the service is not run before luet packages uninstall, so no /run/lock is created. How to check that?

**Comment by geaaru** [comment]
If the package is there then:
1. check if in the current rootfs exists a file under `/etc/tmpfiles.d/*.conf`  or `/usr/lib/tmpfiles.d/*.conf` with the rules for the creation of the directory `/run/lock`. If not, add the file on package spec directory with a step like this: `- cp lock.conf /usr/lib/tmpfiles.d/`
2. check if the systemd service systemd-tmpfiles-setup is active and enabled. If not enable it at live-setup `systemdctl enable systemd-tmpfiles-setup`

The rule must be something like described in the gentoo issue:
`d /run/lock 0775 root uucp -`

**Comment by msdobrescu** [comment]
Thanks, @geaaru, as said above, the file is deployed on the target system. I am waiting for the installer to finish and report the service status in a log.

**Comment by geaaru** [comment]
Fixed with #24
